[sharded tensor] skip empty local tensor in gather

zyan0 opened the pull request:
Summary: local tensor can be empty and cause the gather to fail

Test Plan: jeevg will verify this in a E2E test

Differential Revision: D42103140

wanchaol reviewed (approved):
LGTM, thanks for the fix!

Commit:
[sharded tensor] skip empty local tensor in gather (#91017)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/91017

local tensor can be empty and cause the gather to fail

Test Plan: jeevg will verify this in a E2E test

Reviewed By: wanchaol

Differential Revision: D42103140

fbshipit-source-id: 97dc8a72e1420a691672682958e6fb676e57555c
